backend/api/src/main/kotlin/de/keepmealive3d/adapters/model/ModelDeleteController.kt

```diff
@@ -0,0 +1,41 @@
+package de.keepmealive3d.adapters.model
+
+import de.keepmealive3d.core.auth.KmaUserPrincipal
+import de.keepmealive3d.core.model.ModelRepository
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.auth.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class ModelDeleteController(application: Application): KoinComponent {
+    private val modelRepository: ModelRepository by inject()
+
+    init {
+        application.routing {
+            authenticate("jwt") {
+                delete("/api/model/{filename}") {
+                    val user = call.principal<KmaUserPrincipal>()
+                    val filename = call.parameters["filename"]
+                    if(filename == null) {
```

Comment:
From a security perspective it would be better to first check the login status before the existence of the file.

Reply:
I've changed it, it probably wasn't a huge issue as the only time the user is null is when the token was expired or the user is missing in the db.